main.qml

```diff
@@ -315,16 +315,37 @@ ApplicationWindow {
             webView.url = webView.mainUrl;
         }
     }
+    function injectJS() {
+        webView.webChannel.registerObject( 'transport', transport );
+        // Try-catch to be able to return the error as result, but still throw it in the client context
+        // so it can be caught and reported
+        var injectedJS = "try { initShellComm() } " +
```

Comment:
what would happen if this gets executed twice

Reply:
There are some cases where this may happen. I've tested that even if the code is executed twice, this does not result to any errors and it does not break the normal operation of the shell and the UI.

If the URL is not loaded when the timeout passes. Then the timer will inject the JS code and the following may happen depending on how much of the site is loaded:
* If the scripts are not loaded yet the JS injection from the timer will fail and error message will be shown. When the site loads completely the JS injection will be performed again. This time it should be successful. The user will not be notified.
* if the scripts are loaded successfully but they are corrupted for some reason the error message will be shown twice.
* if the scripts are loaded successfully and working but the connection is not closed for some reason the injected JS from the timer will execute quietly and Stremio will be operational. In that case, when the connection is closed without error, the script will be injected one more time. Of course it will be successful again.